src/ceph-volume/ceph_volume/tests/util/test_device.py

```diff
@@ -150,6 +150,42 @@ def test_is_member_blkid(self, device_info, label):
 
         assert disk.is_member is True
 
+    def test_no_removable_device(self, device_info):
+        data = {"/dev/sdb": {"removable": 1}}
+        device_info(devices=data)
+        disk = device.Device("/dev/sdb")
+        assert not disk.is_valid
+
+    def test_no_removable_device2(self, device_info):
+        data = {"/dev/sdb": {"removable": 0}}
+        device_info(devices=data)
+        disk = device.Device("/dev/sdb")
+        assert disk.is_valid
+
+    def test_no_readonly_device(self, device_info):
```

Comment:
all of these need their counter tests as well

Reply:
ok. done.

Reply:
ok. done.